go-abci-service/abci/monitor.go

```diff
@@ -42,37 +44,77 @@ func (app *AnchorApplication) SyncMonitor() {
 	}
 }
 
-//KeyMonitor : updates active ECDSA public keys from all accessible peers
+//StakeIdentity : updates active ECDSA public keys from all accessible peers
 //Also ensures api is online
-func (app *AnchorApplication) KeyMonitor() {
+func (app *AnchorApplication) StakeIdentity() {
 	for app.JWKSent != true {
 		time.Sleep(60 * time.Second)
+		if _, exists := app.state.CoreKeys[app.ID]; exists {
+			return
+		}
 		if !app.state.ChainSynced {
 			continue
 		}
-		selfStatusURL := fmt.Sprintf("%s/status", app.config.APIURI)
-		response, err := http.Get(selfStatusURL)
+		amValidator, err := app.AmValidator()
 		if app.LogError(err) != nil {
 			continue
 		}
-		contents, err := ioutil.ReadAll(response.Body)
+		//if we're not a validator, we need to "stake" by opening a ln channel to the validators
+		if !amValidator {
+			validators, err := app.rpc.GetValidators(app.state.Height)
```

Comment:
So this ultimately determines how much will need to be staked right? This rpc returns a subset of all validators equal to the required ratio for staking (so if the ratio is 2/3 and there are 15 total cores, this will return 10 validators). And then the actual amount of how much will need to be locked up in channels totals depends on the LocalSats amount for each validator? 

This makes me also wonder if it would make sense to expose somewhere in the API a kind of "dry run" staking where a core can find out exactly how much will be needed to stake before actually running the live command. I'm thinking in particular of situations where channels start to be opened and then they run out of available on chain funds before all the channels are open. This means they have funds locked up in channels but still can't actually participate in anchoring as they haven't opened the requisite number of channels yet and will have to wait for several more blocks while they wait for funds to transfer to their lnd wallet (assuming they have funds available elsewhere) before finishing up. 

It's also possible this is already available somewhere and I just missed it, but just wanted to note it just in case.

Reply:
Yeah, I think a dry-run script might be in order, just to sum up the total needed before an operator has to commit the funds.